Modules/Loadable/Segmentations/Widgets/qMRMLSegmentEditorWidget.cxx

```diff
@@ -1141,6 +1147,13 @@ void qMRMLSegmentEditorWidget::updateEffectList()
     QLayoutItem *child;
     while ((child = d->EffectsGroupBox->layout()->takeAt(0)) != 0);
     }
+  QList<QAbstractButton*> undoRedobuttons = d->UndoRedoButtonGroup.buttons();
+  foreach (QAbstractButton* button, undoRedobuttons)
+    {
+    button->hide();
```

Comment:
Do not show/hide the buttons here. It would interfere with qMRMLSegmentEditorWidget::undoEnabled() or any other low-level customizations that developers might apply (we don't guarantee anything when developers to low-level customizations, but there is also no reason to unnecessarily modify widget properties).

Check undoEnabled method anyway, probably it would be better to show/hide the entire frame instead of show/hide each button.